packages/terra-base/src/Base.jsx

```diff
@@ -48,14 +48,14 @@ class Base extends React.Component {
 
   componentDidMount() {
     if (this.props.locale !== undefined) {
-      i18nLoader(this.props.locale, this.setState, this);
+      i18nLoader(this.props.locale).then(i18n => this.setState(i18n.default));
```

Comment:
I updated this code to use promises... much cleaner and works better with dynamic imports.